exporters/zipkin/src/main/java/io/opentelemetry/exporters/zipkin/ZipkinExporterConfiguration.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.exporters.zipkin;
+
+import com.google.auto.value.AutoValue;
+import javax.annotation.concurrent.Immutable;
+import zipkin2.Span;
+import zipkin2.codec.BytesEncoder;
+import zipkin2.codec.SpanBytesEncoder;
+import zipkin2.reporter.Sender;
+import zipkin2.reporter.urlconnection.URLConnectionSender;
+
+/**
+ * Configurations for {@link ZipkinSpanExporter}.
+ *
+ * @since 0.4.0
+ */
+@AutoValue
+@Immutable
+public abstract class ZipkinExporterConfiguration {
+
+  ZipkinExporterConfiguration() {}
+
+  abstract String getServiceName();
+
+  abstract Sender getSender();
+
+  abstract BytesEncoder<Span> getEncoder();
+
+  /**
+   * Returns a new {@link Builder} with defaults set to an "unknown" serviceName and using the
+   * {@link SpanBytesEncoder#JSON_V2} encoder.
+   *
+   * @return a {@code Builder}.
+   * @since 0.4.0
+   */
+  public static Builder builder() {
+    return new AutoValue_ZipkinExporterConfiguration.Builder()
+        .setEncoder(SpanBytesEncoder.JSON_V2)
+        .setServiceName("unknown");
+  }
+
+  /**
+   * Builds a HTTP exporter for <a href="https://zipkin.io/zipkin-api/#/">Zipkin V2</a> format.
+   *
+   * @param endpoint The Zipkin endpoint URL, ex. "http://zipkinhost:9411/api/v2/spans".
+   * @param serviceName The serviceName with which to identify Spans. See {@link
+   *     Builder#setServiceName(String)} for details.
+   */
+  public static ZipkinExporterConfiguration create(String endpoint, String serviceName) {
```

Comment:
I really don't like factory methods like this personally as two string args can be easily permuted, but I don't feel like holding you up on another round as it isn't important enough and people can figure it out.

Reply:
So... no factory methods, or just the one with a single arg? I have no strong opinion either way.

Reply:
because of "service.name" isn't in a crystal clear state, I think it is better to use the builder similar to how jaeger does, and fail if a fallback service name isn't given. this forces folks to think about this, for example, there's often an application name property somewhere they can reach even if they can't affect instrumentation to affect "service.name" They can be told that if you don't know one set "unknown" but I feel it is better to let them see the importance of it via making the property required.

ex "unknown" makes a almost useless aggregation which can be hard to hunt down. We use this because we used to use it. It is true there's a slight amount of value as you can see bugs where people didn't set a value in one place, and "unknown" is better for people to punt with vs making up new words for "unknown".

All that said, many wouldn't even know to set this unless told to via a required property. I think using required as a means to help users who know a better value set one from config, is a better optimization than what would likely be the opposite, where folks only find out that they should have set it by looking at traces already in the system.